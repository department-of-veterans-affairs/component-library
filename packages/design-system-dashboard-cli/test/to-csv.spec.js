const memfs = require('memfs');
const path = require('path');
const toCSV = require('../src/to-csv');

const json = {
  'dummy.txt': 'Initial dummy placeholder file',
};

jest.mock('fs');
memfs.vol.fromJSON(json, '.');

const flattenedData = [
  {
    date: '2022-04-08',
    component_name: 'alert',
    some_app: 2,
    total: 2,
  },
  {
    date: '2022-04-08',
    component_name: 'accordion',
    some_app: 2,
    another_app: 1,
    total: 3,
  },
];

describe('toCSV', () => {
  it('writes a CSV file', async () => {
    await toCSV(flattenedData, './output.csv');
    const csvPath = path.resolve(__dirname, '../output.csv');
    const csvFile = memfs.vol.toJSON()[csvPath];

    expect(csvFile).toMatchSnapshot();
  });
});
