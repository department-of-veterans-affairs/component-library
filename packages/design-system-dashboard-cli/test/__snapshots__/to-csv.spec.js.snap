// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toCSV writes a CSV file 1`] = `
"date,component_name,some_app,total,another_app
2022-04-08,alert,2,2
2022-04-08,accordion,2,3,1"
`;
